docs/container_images.md

```diff
@@ -1,5 +1,8 @@
 # Container Images
 
+## Binary Authorization
+By default, Kubernetes will pull any container images specified in the `PodSpec` as if they are guaranteed to have been uploaded by the intended person.  However, if your container registry is compromised or the API Server is tricked into pulling the wrong images in any way, this could bring in malicious containers.  One highly recommended approach is to have your container images signed and verified (with a Hash) with an accomapnying policy that enforces this at the cluster level. See [GKE Binary Authorization](https://cloud.google.com/binary-authorization/) for more information.
```

Comment:
`One highly recommended approach is to have your container images signed and verified (with a Hash) with an accomapnying policy that enforces this at the cluster level.` IMO, most organizations should not immediately jump to enforcing signed images.  Instead, they should first restrict/allowlist a specific set of container registries to reduce the majority of the low hanging fruit risk.  As they mature their container build process to be fully centralized, /then/ they would be ready to add image signing during build and enforcement in the cluster.  Otherwise, they'd be breaking deployments and losing trust of their tenants. 
 Thoughts?

Reply:
> Can you clarify what you mean by `as if they are guaranteed to have been uploaded by the intended person`?

Whoops, I took that out, it was ambiguous and not sure what I was trying to get at with that.

>Can you explain the attack vector for the "API Server is tricked into pulling the wrong images in any way"?
That was really bad wording, I think I broke it down better into two attack vectors.  I'm not sure if this is the right way to think about it, let me know.  
Threat 1: Cluster is compromised and attacker is allowed to pull `registry.hub.docker.com/evil-image`, loading a malicious container. Fix: policy that whitelists/blacklists container registries
Threat 2: Container registry is compromised and an existing name of an image `useful-api` is overwritten in the registry, loading a malicious container. Fix: hash digest during the pipeline build of the image

Thanks for pointing that out, it's a lot more clear in my mind now that I've had to type it out and think a little more. Am I missing something else or being too vague ("cluster is compromised" is potentially a little vague, I could say a ServiceAccount or something else if you think necessary)

>IMO, most organizations should not immediately jump to enforcing signed images. Instead, they should first restrict/allowlist a specific set of container registries to reduce the majority of the low hanging fruit risk. As they mature their container build process to be fully centralized, /then/ they would be ready to add image signing during build and enforcement in the cluster. Otherwise, they'd be breaking deployments and losing trust of their tenants.

Agreed, that sounds like practical best steps for implementing this.  I broke it out into another paragraph and tried to speak to that.

Thanks for the feedback and I welcome any more thoughts on the matter!